src/main/java/net/dries007/tfc/util/InteractionManager.java

```diff
@@ -367,10 +367,11 @@ else if (!level.getBlockState(relativePos.below()).isAir())
         }
 
         // Knapping
+        final BiPredicate<ItemStack, Player> rockPredicate = (stack, player) -> TFCConfig.SERVER.requireOffhandForRockKnapping.get() ? Helpers.isItem(player.getMainHandItem(), TFCTags.Items.ROCK_KNAPPING) && Helpers.isItem(player.getOffhandItem(), TFCTags.Items.ROCK_KNAPPING) : stack.getCount() >= 2;
```

Comment:
Honestly, I think having a config option to disable this is kind of dumb, but whatever. I *do* however, think this is executed incorrectly. Default behavior should be to allow off hand *or* two in one stack. As all it cares about is that you have two rocks. This config option (which was requested, so fine, even though I think it's pointless), should just disable one of those.

```java
(offhand && mainhand) || (!require && mainhand >= 2)
```